File: ds-lru-cache/__lab_harness__.mjs
// Harnais généré — ne pas modifier.
const __chunks = [];
const __origWrite = process.stdout.write.bind(process.stdout);
process.stdout.write = (c, e, cb) => { __chunks.push(String(c)); if (typeof e === 'function') e(); else if (typeof cb === 'function') cb(); return true; };
process.stderr.write = (c, e, cb) => { if (typeof e === 'function') e(); else if (typeof cb === 'function') cb(); return true; };
const TESTS = [{"id":"t1","kind":"call-equals","export":"lruProcess","args":[2,[["put",1,10],["put",2,20],["get",1],["put",3,30],["get",2],["get",3]]]},{"id":"t2","kind":"call-equals","export":"lruProcess","args":[3,[["put",5,50],["get",5],["get",9]]]},{"id":"t3","kind":"call-equals","export":"lruProcess","args":[2,[["put",1,1],["put",2,2],["get",1],["put",3,3],["get",1],["get",2]]]},{"id":"t4","kind":"call-equals","export":"lruProcess","args":[2,[["put",1,1],["put",2,2],["put",1,9],["put",3,3],["get",2],["get",1],["get",3]]]}];
(async () => {
  let mod = null, importError = null;
  try { mod = await import("./solution.mjs"); }
  catch (e) { importError = String((e && e.stack) || e).slice(0, 2000); }
  const stdout = __chunks.join('');
  const observed = {};
  for (const t of TESTS) {
    if (t.kind === 'call-equals') {
      const __t0 = Date.now();
      let entry;
      if (importError) entry = { error: importError };
      else {
        try {
          const fn = mod[t.export];
          if (typeof fn !== 'function') entry = { error: "export « " + t.export + " » introuvable ou non-fonction" };
          else { const value = await fn(...(t.args || [])); entry = { value }; }
        } catch (e) { entry = { error: String((e && e.message) || e).slice(0, 500) }; }
      }
      entry.durationMs = Date.now() - __t0;
      observed[t.id] = entry;
    } else {
      observed[t.id] = importError ? { error: importError, stdout } : { stdout };
    }
  }
  const userStdout = __chunks.join('');
  process.stdout.write = __origWrite;
  let payload;
  try { payload = JSON.stringify({ observed, stdout: userStdout.slice(0, 100000) }); }
  catch { payload = JSON.stringify({ observed: {}, fatal: "résultat non sérialisable" }); }
  __origWrite("__LAB_RESULT__" + payload + "\n");
})();

File: ml-leakage-temporal/__lab_harness__.py
# Harnais genere - ne pas modifier.
import sys, json, io, importlib.util, traceback
TESTS = json.loads("[{\"id\":\"t1\",\"kind\":\"call-equals\",\"export\":\"suspect_leaks\",\"args\":[[{\"age\":20,\"refund\":1,\"y\":1},{\"age\":30,\"refund\":0,\"y\":0},{\"age\":40,\"refund\":1,\"y\":1},{\"age\":50,\"refund\":0,\"y\":0}],\"y\"]},{\"id\":\"t2\",\"kind\":\"call-equals\",\"export\":\"suspect_leaks\",\"args\":[[{\"age\":20,\"score\":3,\"y\":1},{\"age\":30,\"score\":9,\"y\":0},{\"age\":40,\"score\":1,\"y\":1},{\"age\":50,\"score\":8,\"y\":0}],\"y\"]},{\"id\":\"t3\",\"kind\":\"call-equals\",\"export\":\"suspect_leaks\",\"args\":[[{\"a\":1,\"b\":1,\"y\":1},{\"a\":0,\"b\":0,\"y\":0},{\"a\":1,\"b\":1,\"y\":1},{\"a\":0,\"b\":0,\"y\":0}],\"y\"]}]")
ENTRY = "solution.py"
MARKER = "__LAB_RESULT__"
_buf = io.StringIO()
_real = sys.stdout
sys.stdout = _buf
sys.stderr = io.StringIO()
mod = None
import_error = None
try:
    spec = importlib.util.spec_from_file_location("user_entry", ENTRY)
    mod = importlib.util.module_from_spec(spec)
    spec.loader.exec_module(mod)
except Exception:
    import_error = traceback.format_exc()[:2000]
observed = {}
import time as _time
for t in TESTS:
    tid = t["id"]
    if t["kind"] == "call-equals":
        _t0 = _time.time()
        if import_error:
            entry = {"error": import_error}
        else:
            fn = getattr(mod, t["export"], None)
            if not callable(fn):
                entry = {"error": "export '%s' introuvable ou non-callable" % t["export"]}
            else:
                try:
                    entry = {"value": fn(*t.get("args", []))}
                except Exception as e:
                    entry = {"error": (repr(e))[:500]}
        entry["durationMs"] = int((_time.time() - _t0) * 1000)
        observed[tid] = entry
    else:
        observed[tid] = {"error": import_error, "stdout": _buf.getvalue()} if import_error else {"stdout": _buf.getvalue()}
user_stdout = _buf.getvalue()[:100000]
sys.stdout = _real
try:
    payload = json.dumps({"observed": observed, "stdout": user_stdout})
except Exception:
    payload = json.dumps({"observed": {}, "fatal": "resultat non serialisable"})
sys.stdout.write(MARKER + payload + "\n")
